static/src/javascripts/projects/commercial/modules/close-disabled-slots.js

```diff
@@ -1,23 +1,38 @@
 // @flow
 import qwery from 'qwery';
 import fastdom from 'lib/fastdom-promise';
+import { commercialFeatures } from 'commercial/modules/commercial-features';
 
 const adSlotSelector: string = '.js-ad-slot';
+const mpuCandidateClass: string = 'fc-slice__item--mpu-candidate';
+const mpuHiderClass: string = 'fc-slice__item--no-mpu';
+const mpuCandidateSelector: string = `.${mpuCandidateClass}`;
+
+const shouldDisableAdSlotWhenAdFree = adSlot =>
+    commercialFeatures.adFree &&
+    (adSlot.className.toLowerCase().includes(mpuCandidateClass) ||
+        !adSlot.className.toLowerCase().includes('merchandising'));
 
 const shouldDisableAdSlot = adSlot =>
-    window.getComputedStyle(adSlot).display === 'none';
+    window.getComputedStyle(adSlot).display === 'none' ||
+    shouldDisableAdSlotWhenAdFree(adSlot);
 
 const closeDisabledSlots = (force: boolean): Promise<void> => {
     // Get all ad slots
     let adSlots: Array<Element> = qwery(adSlotSelector);
+    let mpuCandidates: Array<Element> = qwery(mpuCandidateSelector);
 
     if (!force) {
         // remove the ones which should not be there
         adSlots = adSlots.filter(shouldDisableAdSlot);
+        mpuCandidates = mpuCandidates.filter(shouldDisableAdSlot);
```

Comment:
I've switched the `or` logic, as suggested. `mpuCandidate` slots were never previously "closed" and under ad-free, they don't get loaded either, which was leaving the empty space in the page where they're found (you can see this in PROD at the moment if you're an ad-free trial user). This code makes sure they're properly hidden as the page re-flows those elements to properly fill the otherwise empty space. We don't have many `mpuCandidate`s in any given page, so the overhead shouldn't be excessive.